frontend/src/components/explorer/MapViewer.vue

```diff
@@ -277,10 +277,20 @@ export default {
       const queryString = Object.entries(newQuery)
         .map(e => e.join('='))
         .join('&');
-      const payload = [{}, null, `${this.$route.path}?${queryString}`];
+
       if (newQuery.dim === this.$route.query.dim || (newQuery.dim && !this.$route.query.dim)) {
+        const payload = [{}, null, `${this.$route.path}?${queryString}`];
         history.replaceState(...payload); // eslint-disable-line no-restricted-globals
       } else {
+        let urlPath = `${this.$route.path}`;
```

Comment:
To avoid future confusion, I think a comment about what's going on would be nice here

Reply:
Thanks for the comment! I've added some explanation to the code in the commit 3722c94

However, I'm not quite sure to which extent the code should be commented. If you go through the code for MetAtlas, there are very little comment. It's great that you raised this question and I think we should discuss about this: whether and how should we add comment in the code.

Reply:
Thanks! And yes, I have noticed the lack of comments :). I think at least adding very short comments would help in many places, the code base is quite massive and it's sometimes hard to figure out what's going on. But there are of course drawbacks of comments, such as the risk of them not being updated when the code is, etc. So es, a discussion would be good!